src/core_modules/capture-core/metaDataStoreLoaders/trackedEntityTypes/loadTrackedEntityTypes.js

```diff
@@ -1,42 +1,32 @@
 // @flow
-import StorageController from 'capture-core-utils/storage/StorageController';
-import { metaTrackedEntityTypesApiSpecification, trackedEntityTypesApiSpecification } from '../../api';
-import getTrackedEntityTypesLoadSpecification
-    from '../../apiToStore/loadSpecifications/getTrackedEntityTypesLoadSpecification';
-
-function getOptionSetsMeta(trackedEntityTypesMeta) {
-    if (!trackedEntityTypesMeta) {
-        return [];
-    }
+import { queryTrackedEntityTypesOutline } from './queries';
+import { storeTrackedEntityTypes } from './quickStoreOperations';
 
-    const optionSetsMeta = trackedEntityTypesMeta.reduce((accOptionSetsMeta, type) => {
-        const optionSetsMetaForType = type.trackedEntityTypeAttributes &&
-            type.trackedEntityTypeAttributes.reduce((accOptionSetsMetaForType, TETA) => {
+function getOptionSetsOutline(trackedEntityTypesOutline) {
+    const optionSetsOutline = trackedEntityTypesOutline.reduce((accOptionSets, type) => {
+        const optionSetsForType = type.trackedEntityTypeAttributes &&
+            type.trackedEntityTypeAttributes.reduce((accOptionSetsForType, TETA) => {
                 if (TETA.trackedEntityAttribute && TETA.trackedEntityAttribute.optionSet) {
-                    accOptionSetsMetaForType.push(TETA.trackedEntityAttribute.optionSet);
+                    accOptionSetsForType.push(TETA.trackedEntityAttribute.optionSet);
                 }
-                return accOptionSetsMetaForType;
+                return accOptionSetsForType;
             }, []);
 
-        if (optionSetsMetaForType) {
-            accOptionSetsMeta = [...accOptionSetsMeta, ...optionSetsMetaForType];
+        if (optionSetsForType) {
+            accOptionSets = [...accOptionSets, ...optionSetsForType];
```

Comment:
This could have been better, but then again I was just renaming some variables.

Reply:
LGTM gave me an error so I looked into it. It turns out I've refactored this in the improvement part of the issue, so this will be completely different in the last PR.

Reply:
This is been followed up in the last PR. I will leave this unresolved to keep a note when this is PR is approved. 

++

Reply:
I see you fixes this in this PR, is that correct?

Reply:
Totally refactored in last PR (Just fixed a bug here)